### ISSUE TITLE
Template expressions are incorrectly indented

### ISSUE THREAD

**Issue by kassens** [benjamn/recast#216]
If a template expression is printed that's indented, it seems like indentation is added twice somewhere. This failing test illustrates the bug.

```javascript
it("preserves indentation in template expressions", function() {
    var printer = new Printer({
        tabWidth: 2
    });
    code = [
        "var x = {",
        "  y: () => Relay.QL`",
        "    query {",
        "      ${foo},",
        "      field,",
        "    }",
        "  `",
        "};",
    ].join("\n");

    ast = parse(code);
    pretty = printer.printGenerically(ast).code;
    assert.strictEqual(pretty, code);
});
```

Output
```diff
-actual
+expected

 var x = {
   y: () => Relay.QL`
-      query {
-        ${foo},
-        field,
-      }
-    `
+    query {
+      ${foo},
+      field,
+    }
+  `
 };
```

**Comment by kassens** [comment]
cc @fkling, @cpojer

**Comment by benjamn** [comment]
Oh, interesting. I'm not sure the spaces inside the back-ticks are even considered as indentation, as far as Recast is currently concerned. So part of the double indentation is coming from the pretty printer, and the other half is coming from the original template string expression.

GraphQL template strings can be reindented harmlessly, but I think in general we need to avoid adding or removing "indentation" from multi-line template strings.

**Comment by kassens** [comment]
A heuristic that could be used: if all lines are indented, assume it's some sort of markup and okay to re-indent.

Preserving the same indent would already be better and solve most cases (when changes don't affect indentation).

**Comment by yungsters** [comment]
Started looking into this again: http://felix-kling.de/esprima_ast_explorer/#/N2D8hbBQ3I/1

**Comment by kassens** [comment]
Thanks!

**Comment by yungsters** [comment]
Looks like this still exhibits a bug after 9e6fdb8: http://felix-kling.de/esprima_ast_explorer/#/N2D8hbBQ3I/1

**Comment by yungsters** [comment]
@benjamn Should I file a new issue for this?

**Comment by benjamn** [comment]
This was still broken due to a combination of incomplete indentation-locking logic (4de9390f85666d7707c67b7970943d2e97548d6b) and some bad `.loc` information for `TemplateElement` nodes (f45c3c0c63bbe6c2717768ca181d3390e1a5ac39).

Just published [v0.10.39](https://github.com/benjamn/recast/releases/tag/v0.10.39) to NPM: https://www.npmjs.com/package/recast

**Comment by nickretallack** [comment]
This is still broken.  I originally filed an issue here: https://github.com/zxbodya/flowts/issues/8

It seems to be looking at the first line of the template string and stripping that much indentation from all later lines.
